### PR TITLE
Deprecate EBSDMesh, add EBSDMeshGenerator

### LINKED ISSUE #16062: Transition EBSDMesh to the MeshGenerator system.
## Reason
To enable EBSD meshes to work with MeshGenerators we need to transition to that system.

## Design
Make this  amesh generator. Should be straight forward.

## Impact
Updated capability.

### PR DESCRIPTION
**Pull request by dschwen**
Closes #16062 

Helpful deprecation message:

```
examples/ebsd_reconstruction.IN100-111grn: *** ERROR ***
examples/ebsd_reconstruction.IN100-111grn:
examples/ebsd_reconstruction.IN100-111grn:
examples/ebsd_reconstruction.IN100-111grn: Deprecated code:
examples/ebsd_reconstruction.IN100-111grn: EBSDMesh is deprecated, please use the EBSDMeshGenerator instead. For example:
examples/ebsd_reconstruction.IN100-111grn:
examples/ebsd_reconstruction.IN100-111grn: [Mesh]
examples/ebsd_reconstruction.IN100-111grn:   type = EBDSMesh
examples/ebsd_reconstruction.IN100-111grn:   filename = my_ebsd_data.dat
examples/ebsd_reconstruction.IN100-111grn: []
examples/ebsd_reconstruction.IN100-111grn:
examples/ebsd_reconstruction.IN100-111grn: becomes
examples/ebsd_reconstruction.IN100-111grn:
examples/ebsd_reconstruction.IN100-111grn: [Mesh]
examples/ebsd_reconstruction.IN100-111grn:   [ebsd_mesh]
examples/ebsd_reconstruction.IN100-111grn:     type = EBDSMeshGenerator
examples/ebsd_reconstruction.IN100-111grn:     filename = my_ebsd_data.dat
examples/ebsd_reconstruction.IN100-111grn:   []
examples/ebsd_reconstruction.IN100-111grn: []
examples/ebsd_reconstruction.IN100-111grn:
```

### CODE REVIEW

**Review by aeslaughter — DISMISSED** [review]

File: modules/phase_field/src/meshgenerators/EBSDMeshGenerator.C [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+registerMooseObject("PhaseFieldApp", EBSDMeshGenerator);
+
+InputParameters
+EBSDMeshGenerator::validParams()
+{
+  InputParameters params = GeneratedMeshGenerator::validParams();
+  params.addClassDescription("Mesh generated from a specified DREAM.3D EBSD data file.");
+  params.addRequiredParam<FileName>("filename", "The name of the file containing the EBSD data");
+  params.addParam<unsigned int>(
```
Comment (aeslaughter) [inline modules/phase_field/src/meshgenerators/EBSDMeshGenerator.C:23]: Why do you have this? It is already available in Mesh/uniform_refine

Comment (dschwen) [reply modules/phase_field/src/meshgenerators/EBSDMeshGenerator.C:23]: No. Mit with that updated documentation.

File: modules/phase_field/include/meshgenerators/EBSDMeshGenerator.h [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  // Interface functions for the EBSDReader
+  const Geometry & getEBSDGeometry() const { return _geometry; }
+  const std::string & getEBSDFilename() const { return _filename; }
+
+protected:
+  /// Read the EBSD data file header
+  void readEBSDHeader();
+
+  /// Name of the file containing the EBSD data
+  std::string _filename;
```
[the remark this answers is not in the document]
Comment (dschwen) [reply modules/phase_field/include/meshgenerators/EBSDMeshGenerator.h:49]: Isn't there a derived string type that should be used for this?

File: modules/phase_field/src/meshgenerators/EBSDMeshGenerator.C (hunk 1 above)
Comment (aeslaughter) [reply modules/phase_field/src/meshgenerators/EBSDMeshGenerator.C:23]: Can you use setDocString instead: https://github.com/idaholab/moose/blob/01b19be81e97d49e48b700c787c652322a8e6459/framework/src/utils/InputParameters.C#L297

Comment (aeslaughter) [reply modules/phase_field/src/meshgenerators/EBSDMeshGenerator.C:23]: No you can't, you use it in here.

File: modules/phase_field/include/meshgenerators/EBSDMeshGenerator.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (aeslaughter) [reply modules/phase_field/include/meshgenerators/EBSDMeshGenerator.h:49]: Yes. Let me fix this and the stupid commit I made, which is wrong the cause the uniform_refine parameter should come from this class. I will push a new commit.

**Review by aeslaughter — APPROVED** [review]

**Pull request merged**